### PR TITLE
split sig-release testgrids to own subdir

### PR DESCRIPTION
**Pull request by spiffxp**
specifically:
- use OWNERS file from config/jobs/kubernetes/sig-release
- alphasort dashboard names
- move presubmits-release-notes to sig-release-release-notes-presubmits

/sig release
/priority important-soon

### CODE REVIEW

**Comment by spiffxp** [comment]
/cc onyiny-ang jeefy marpaia
release-notes root owners

File: config/testgrids/kubernetes/sig-release/OWNERS [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
+- calebamiles
+- justaugustus
+- tpepper
+
+labels:
+- sig/release
```
Comment (justaugustus) [inline config/testgrids/kubernetes/sig-release/OWNERS]:
```suggestion
- sig/release
- area/release-eng
```

Comment (spiffxp) [reply config/testgrids/kubernetes/sig-release/OWNERS]: Done, and added this label to config/jobs/kubernetes/sig-release/ as well so the OWNERS files match

**Comment by justaugustus** [comment]
You're a saint, @spiffxp! :heart: 
/lgtm
/approve
/priority important-soon
/milestone v1.16
/area release-eng

**Pull request merged**